lib/workos/organization.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+# typed: true
+
+require 'net/http'
+
+module WorkOS
+  # The Organization module provides resource methods for working with
+  # organizations, which are the organizations that have connections configured
+  # within your WorkOS dashboard.
+  module Organization
+    class << self
+      extend T::Sig
+      include Base
+      include Client
+
+      # Retrieve a list of organizations
+      #
+      # @param [String] domain Filter organizations to only return those that
+      #  are associated with the provided domain.
+      # @param [String] before A pagination argument used to request
+      #  organizations before the provided Organization ID.
+      # @param [String] after A pagination argument used to request
+      #  organizations after the provided Organization ID.
+      # @param [Integer] limit A pagination argument used to limit the number
+      #  of listed Organizations that are returned.
+      sig do
+        params(
+          options: T::Hash[Symbol, String],
+        ).returns(WorkOS::Types::ListStruct)
+      end
+      # rubocop:disable Metrics/MethodLength
+      def list(options = {})
```

Comment:
I know we didn't land on a name so I dropped this in temporarily. Happy to adjust to the find decided name!

:yak_shaving: